sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -104,8 +111,40 @@ private object DateTimeFormatterHelper {
       .withResolverStyle(ResolverStyle.STRICT)
   }
 
-  def buildFormatter(pattern: String, locale: Locale): DateTimeFormatter = {
-    val builder = createBuilder().appendPattern(pattern)
+  /**
+   * Building a formatter for parsing seconds fraction with variable length
+   */
+  def createBuilderWithVarLengthSecondFraction(
+      pattern: String,
+      builder: DateTimeFormatterBuilder): DateTimeFormatterBuilder = {
```

Comment:
nit: do we need this parameter? I think it's safer to create a fresh builder inside this method.